--- a/src/rc/openrc-init.c
+++ b/src/rc/openrc-init.c
@@ -53,7 +53,7 @@ static pid_t do_openrc(const char *runlevel)
 			sigemptyset(&signals);
 			sigprocmask(SIG_SETMASK, &signals, NULL);
 			printf("Starting %s runlevel\n", runlevel);
-			execl("/sbin/openrc", "/sbin/openrc", runlevel, NULL);
+			execl("/bin/openrc", "/bin/openrc", runlevel, NULL);
 			perror("exec");
 			break;
 		default:
--- a/src/rc/rc-selinux.c
+++ b/src/rc/rc-selinux.c
@@ -400,8 +400,8 @@ void selinux_setup(char **argv)
 	 * which will open the pty with initrc_devpts_t, if it doesnt exist,
 	 * fall back to plain exec
 	 */
-	if (!access("/usr/sbin/open_init_pty", X_OK)) {
-		if (execvp("/usr/sbin/open_init_pty", argv)) {
+	if (!access("/bin/open_init_pty", X_OK)) {
+		if (execvp("/bin/open_init_pty", argv)) {
 			perror("execvp");
 			exit(-1);
 		}
--- a/src/rc/rc.c
+++ b/src/rc/rc.c
@@ -78,8 +78,8 @@ const char *usagestring = ""					\
 #define INITSH                  RC_LIBEXECDIR "/sh/init.sh"
 #define INITEARLYSH             RC_LIBEXECDIR "/sh/init-early.sh"
 
-#define SHUTDOWN                "/sbin/shutdown"
-#define SULOGIN                 "/sbin/sulogin"
+#define SHUTDOWN                "/bin/shutdown"
+#define SULOGIN                 "/bin/sulogin"
 
 #define INTERACTIVE             RC_SVCDIR "/interactive"
 
@@ -288,8 +288,8 @@ open_shell(void)
 	/* VSERVER systems cannot really drop to shells */
 	if (sys && strcmp(sys, RC_SYS_VSERVER) == 0)
 	{
-		execl("/sbin/halt", "/sbin/halt", "-f", (char *) NULL);
-		eerrorx("%s: unable to exec `/sbin/halt': %s",
+		execl("/bin/halt", "/bin/halt", "-f", (char *) NULL);
+		eerrorx("%s: unable to exec `/bin/halt': %s",
 		    applet, strerror(errno));
 	}
 #endif
