diff --git a/auto/feature.old b/auto/feature
index 3561f59..d2cf977 100644
--- a/auto/feature
+++ b/auto/feature
@@ -50,55 +50,6 @@ eval "/bin/sh -c \"$ngx_test\" >> $NGX_AUTOCONF_ERR 2>&1"
 if [ -x $NGX_AUTOTEST ]; then
 
     case "$ngx_feature_run" in
-
-        yes)
-            # /bin/sh is used to intercept "Killed" or "Abort trap" messages
-            if /bin/sh -c $NGX_AUTOTEST >> $NGX_AUTOCONF_ERR 2>&1; then
-                echo " found"
-                ngx_found=yes
-
-                if test -n "$ngx_feature_name"; then
-                    have=$ngx_have_feature . auto/have
-                fi
-
-            else
-                echo " found but is not working"
-            fi
-        ;;
-
-        value)
-            # /bin/sh is used to intercept "Killed" or "Abort trap" messages
-            if /bin/sh -c $NGX_AUTOTEST >> $NGX_AUTOCONF_ERR 2>&1; then
-                echo " found"
-                ngx_found=yes
-
-                cat << END >> $NGX_AUTO_CONFIG_H
-
-#ifndef $ngx_feature_name
-#define $ngx_feature_name  `$NGX_AUTOTEST`
-#endif
-
-END
-            else
-                echo " found but is not working"
-            fi
-        ;;
-
-        bug)
-            # /bin/sh is used to intercept "Killed" or "Abort trap" messages
-            if /bin/sh -c $NGX_AUTOTEST >> $NGX_AUTOCONF_ERR 2>&1; then
-                echo " not found"
-
-            else
-                echo " found"
-                ngx_found=yes
-
-                if test -n "$ngx_feature_name"; then
-                    have=$ngx_have_feature . auto/have
-                fi
-            fi
-        ;;
-
         *)
             echo " found"
             ngx_found=yes
