--- a/Makefile
+++ b/Makefile
@@ -4,20 +4,20 @@ ifeq ($(VERBOSE),0)
 MAKEFLAGS += --no-print-directory
 endif
 
-PREFIX?=/usr
-LIBDIR?=$(PREFIX)/lib
-SBINDIR?=/sbin
-CONFDIR?=/etc/iproute2
-DATADIR?=$(PREFIX)/share
-HDRDIR?=$(PREFIX)/include/iproute2
+PREFIX=
+LIBDIR=/lib
+SBINDIR=/bin
+CONFDIR=/etc/iproute2
+DATADIR=/share
+HDRDIR=/include/iproute2
 DOCDIR?=$(DATADIR)/doc/iproute2
 MANDIR?=$(DATADIR)/man
 ARPDDIR?=/var/lib/arpd
-KERNEL_INCLUDE?=/usr/include
+KERNEL_INCLUDE=$(SYSROOT)/include
 BASH_COMPDIR?=$(DATADIR)/bash-completion/completions
 
 # Path to db_185.h include
-DBM_INCLUDE:=$(DESTDIR)/usr/include
+DBM_INCLUDE:=$(SYSROOT)/include
 
 SHARED_LIBS = y
 
